banzai/utils/file_utils.py

```diff
@@ -85,4 +132,31 @@ def get_calibration_filename(image):
                 cal_file += '-{}'.format(filename_part)
         cal_file += '.fits'
         return cal_file
+
     return get_calibration_filename
+
+
+def get_basename(path):
+    basename = None
+    if path is not None:
+        filename = os.path.basename(path)
+        if filename.find('.') > 0:
+            basename = filename[:filename.index('.')]
+        else:
+            basename = filename
+    return basename
+
+
+def download_from_s3(file_info, output_directory, runtime_context):
+    frame_id = file_info.get('frameid')
+    if frame_id is not None:
+        url = f'{runtime_context.ARCHIVE_FRAME_URL}/{frame_id}'
+    else:
+        basename = get_basename(file_info.get('path'))
+        url = f'{runtime_context.ARCHIVE_FRAME_URL}/?basename={basename}'
```

Comment:
I think this returns a list instead of a single element so make sure we cover that case.